crates/ra_ide_api/src/imp.rs

```diff
@@ -243,21 +247,74 @@ impl db::RootDatabase {
         &self,
         position: FilePosition,
         new_name: &str,
-    ) -> Cancelable<Vec<SourceFileEdit>> {
-        let res = self
-            .find_all_refs(position)?
-            .iter()
-            .map(|(file_id, text_range)| SourceFileEdit {
-                file_id: *file_id,
+    ) -> Cancelable<Option<SourceChange>> {
+        let mut source_file_edits = Vec::new();
+        let mut file_system_edits = Vec::new();
+
+        let source_file = self.source_file(position.file_id);
+        let syntax = source_file.syntax();
+        // We are rename a mod
+        if let (Some(ast_module), Some(name)) = (
+            find_node_at_offset::<ast::Module>(syntax, position.offset),
+            find_node_at_offset::<ast::Name>(syntax, position.offset),
```

Comment:
I think it would be more robust to find a `Name`, and then check that its parent is `Module`. Otherwise, we might find a name inside a module, but not the name of the module:

```
mod m {
   fn foo<|>() {}
}
```